File: CFOnlineShop/CFOnlineShop/ViewController/CFHomePageController/view/CFHomeCollectionHeader.m
//
//  CFHomeCollectionHeader.m
//  CFOnlineShop
//
//  Created by chenfeng on 2018/7/19.
//  Copyright © 2018年 chenfeng. All rights reserved.
//

#import "CFHomeCollectionHeader.h"
#import "SDCycleScrollView.h"

@implementation CFHomeCollectionHeader

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
        
        NSArray *array = @[@"advertisement_1",@"advertisement_2",@"advertisement_3"];
        //本地加载图片的轮播器
        SDCycleScrollView *cycleScrollView = [SDCycleScrollView cycleScrollViewWithFrame:self.frame imageNamesGroup:array];
        cycleScrollView.backgroundColor = kWhiteColor;
        cycleScrollView.autoScrollTimeInterval = 3;
        cycleScrollView.bannerImageViewContentMode = UIViewContentModeScaleAspectFit;
        cycleScrollView.currentPageDotColor = kRedColor;
        [self addSubview:cycleScrollView];
        
    }
    return self;
}

@end

File: SwiftOnlineShopDemo/SwiftOnlineShopDemo/ViewController/CFHomePageController/view/CFHomeCollectionHeader.swift
//
//  CFHomeCollectionHeader.swift
//  SwiftOnlineShopDemo
//
//  Created by chenfeng on 2019/4/25.
//  Copyright © 2019年 chenfeng. All rights reserved.
//

import UIKit

class CFHomeCollectionHeader: UICollectionReusableView {
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        
        let array = ["advertisement_1","advertisement_2","advertisement_3"];
        
        let cycleScrollView = SDCycleScrollView(frame: frame, imageNamesGroup: array)
        
        cycleScrollView?.backgroundColor = UIColor.white;
        cycleScrollView?.autoScrollTimeInterval = 3;
        cycleScrollView?.bannerImageViewContentMode = .scaleAspectFit;
        cycleScrollView?.currentPageDotColor = UIColor.red;
        self.addSubview(cycleScrollView!)
        
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    
}
